### PR TITLE
feat: add has verified factor mfa claim

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

Supercedes #1823  and adds a `HasVerifiedFactor` claim so that it's easier to enforce aal2 only if MFA is enabled on a user's account. Naming is open for discussion

## Additional context

Internally, prior to deploy, we will need to update tests (if any) to reflect that we have added a new claim. As this is an additive change, it should not break any existing setups

### CODE REVIEW

File: internal/hooks/auth_hooks.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ type AccessTokenClaims struct {
 	AuthenticatorAssuranceLevel   string                 `json:"aal,omitempty"`
 	AuthenticationMethodReference []models.AMREntry      `json:"amr,omitempty"`
 	SessionId                     string                 `json:"session_id,omitempty"`
+	HasVerifiedFactor             bool                   `json:"has_verified_factor"`
```
Comment (J0) [inline internal/hooks/auth_hooks.go:111]: Should be set to false if there are no factors / nil slice so there's no need for an omitempty specifier

**Review by kangmingtay — APPROVED** [review]
@J0 can we also add a test to check that the value in the claim is expected?

File: internal/hooks/auth_hooks.go (hunk 1 above)
Comment (kangmingtay) [inline internal/hooks/auth_hooks.go:111]: should we call this `mfa_enabled` ? seems easier to understand vs `has_verified_factor`?

**Comment by J0** [comment]
Currently we don't use `amr` values as per RFC 8176 values. Could look into switching but think that might be part of a bigger discussion

The above convention would work though though we'd need to make an exception for `totp` as it's  not prefixed with `mfa/`

MFA methods:

`totp`
`mfa/webauthn`
`mfa/phone`

**Comment by J0** [comment]
Closing in favour of documentation, should evaluate RFC8176 at some point

**Pull request closed without merging**